### PR TITLE
Include binary wheels in release

### PR DESCRIPTION
**Pull request by levand**
This PR builds binary wheels for the following platforms:

- Windows amd64 (AMD & Intel processors)
- Linux amd64 and arm64 (manylinux)
- Intel and Apple Silicon OSX

Note that 32 bit binaries are not included, as they caused the build to take 10x as long due to the emulation required. Distributions are also only for standard CPython, PyPy binaries are not included. Most linux distros will work with the manylinux build but a few (e.g Alpine) will not.

See the readme for instructions on how to rebuild for AVX support and what the release process for this repo is. 

Note that the code to deploy to production PyPI is commented out, it will need to be enabled after adding the appropriate token to this repo's secrets.

Fixes https://github.com/chroma-core/chroma/issues/885 and https://github.com/chroma-core/chroma/issues/993

### CODE REVIEW

**Comment by levand** [comment]
You can observe example build artifacts here:

https://test.pypi.org/project/chroma-hnswlib/0.7.3a6/#files

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      uses: actions/download-artifact@v3
+      with:
+        name: python-package-distributions
+        path: dist/
+    - name: Publish to Test PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+        repository-url: https://test.pypi.org/legacy/
+    # - name: Publish to PyPI
```
Comment (HammadB) [inline .github/workflows/release.yml:66]: thanks

Comment (HammadB) [reply .github/workflows/release.yml:66]: Added PYPI_API_TOKEN, we can uncomment now.

File: setup.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 from setuptools import Extension, setup
 from setuptools.command.build_ext import build_ext
 
-__version__ = '0.7.2'
+__version__ = '0.7.3.alpha6'
```
Comment (HammadB) [inline setup.py:11]: Just to confirm, idea is to keep version tag and this file in lockstep for now ?

(I can change this to use setuptools_scm later)

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        uses: docker/setup-qemu-action@v2
+        if: matrix.os == 'ubuntu-latest'
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Install cibuildwheel
+        run: python -m pip install cibuildwheel==2.15.0
+      - name: Build wheels
+        run: python -m cibuildwheel --output-dir dist
```
Comment (HammadB) [inline .github/workflows/release.yml:30]: nice find on cibuildwheel

**Review by HammadB — APPROVED** [review]

File: setup.py (hunk 2 above)
Comment (levand) [reply setup.py:11]: Yes, that seemed the simplest thing for now. Could definitely do a more elaborate setup similar to `chroma` but might not even be necessary if this isn't being updated as frequently.

File: .github/workflows/release.yml (hunk 1 above)
Comment (levand) [reply .github/workflows/release.yml]: done

**Pull request merged**